hail/python/hail/methods/statgen.py

```diff
@@ -1480,13 +1480,23 @@ def pca(entry_expr, k=10, compute_loadings=False) -> Tuple[List[float], Table, T
         field = Env.get_uid()
         mt = mt.select_entries(**{field: entry_expr})
     mt = mt.select_cols().select_rows().select_globals()
-
-    r = Env.hail().methods.PCA.apply(mt._jmt, field, k, compute_loadings)
-    scores = Table._from_java(Env.hail().methods.PCA.scoresTable(mt._jmt, r._2()))
-    loadings = from_option(r._3())
-    if loadings:
-        loadings = Table._from_java(loadings)
-    return jiterable_to_list(r._1()), scores, loadings
+    
+    t = Table(MatrixToTableApply(mt._mir, {
+        'name': 'PCA',
+        'entryField': field,
+        'k': k,
+        'computeLoadings': compute_loadings
+    }))
```

Comment:
Where was the old persist?  I'm not seeing it.  U (if computed) is a breeze matrix time the original RowMatrix which is cached.

Reply:
the old persist was done implicitly because it created TableLiterals.

Changing to a relational function will make everything lazy - so in this case, you'd eagerly evaluate PCA once to return eigenvalues, and scores and loadings will evaluate as well.